### PR TITLE
[SCB-775]support invoke service using raw type like JsonObject & String

### PR DESCRIPTION
**Pull request by liubao68**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    super();
+  }
+
+  @Override
+  public <T> T convertValue(Object fromValue, JavaType toValueType) throws IllegalArgumentException {
+    if (String.class.isInstance(fromValue)
+        && !String.class.equals(toValueType.getRawClass())
+        && !java.util.Date.class.equals(toValueType.getRawClass())
+        && !ClassUtils.isPrimitiveOrWrapper(toValueType.getRawClass())) {
+      return super.convertValue(new JsonObject((String) fromValue), toValueType);
```
Comment (wujimin) [inline demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: try this:
super.readValue((String)fromValue, toValueType)

maybe also ok.

Comment (wujimin) [reply demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: about toValueType, try org.springframework.beans.BeanUtils#isSimpleValueType?

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/RestObjectMapper.java [hunk 2]
```diff
@@ -17,25 +17,37 @@
...
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
 import com.fasterxml.jackson.databind.type.TypeFactory;
 
-public final class RestObjectMapper extends ObjectMapper {
-  public static final RestObjectMapper INSTANCE = new RestObjectMapper();
+import io.vertx.core.json.JsonObject;
+
+public class RestObjectMapper extends AbstractRestObjectMapper {
+  private static class JsonObjectSerializer extends JsonSerializer<JsonObject> {
```
Comment (wujimin) [inline common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/RestObjectMapper.java]: seems someone already use field "INSTANCE"
so just add a setter method?

Comment (liubao68) [reply common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/RestObjectMapper.java]: This code looks bad. And I think if others use this, they can change it easily as this PR has done. So it's not a problem.

File: demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java [hunk 3]
```diff
@@ -56,7 +60,11 @@ public RestObjectMapperWithStringMapper() {
   public <T> T convertValue(Object fromValue, JavaType toValueType) throws IllegalArgumentException {
     if (String.class.isInstance(fromValue)
         && !BeanUtils.isSimpleValueType(toValueType.getRawClass())) {
-      return super.convertValue(new JsonObject((String) fromValue), toValueType);
+      try {
+        return super.readValue((String) fromValue, toValueType);
+      } catch (IOException e) {
+        LOGGER.error("Failed to convert value for {}.", e.getMessage());
```
Comment (wujimin) [inline demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: go on or throw?

File: demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java [hunk 4]
```diff
@@ -43,6 +45,8 @@
  *  conversion change. You must write  convertValue to check possible types using.
  */
 public class RestObjectMapperWithStringMapper extends RestObjectMapper {
+  private static Logger LOGGER = LoggerFactory.getLogger(RestObjectMapperWithStringMapper.class);
+
   private static final long serialVersionUID = -8158869347066287575L;
 
   private static final JavaType STRING_JAVA_TYPE = TypeFactory.defaultInstance().constructType(String.class);
```
Comment (wujimin) [inline demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: declare but not used?

File: demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java [hunk 5]
```diff
@@ -43,6 +45,8 @@
  *  conversion change. You must write  convertValue to check possible types using.
  */
 public class RestObjectMapperWithStringMapper extends RestObjectMapper {
```
Comment (wujimin) [inline demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: put this demonstrate class in demo project or treat it as a product class?

Comment (liubao68) [reply demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: This is a demo

File: demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java (hunk 3 above)
Comment (liubao68) [reply demo/demo-schema/src/main/java/org/apache/servicecomb/demo/RestObjectMapperWithStringMapper.java]: I mean go on. Seems better.

**Pull request merged**